src/aws-manager.js

```diff
@@ -242,10 +242,111 @@ class AwsManager {
     this.reportAmiUsage = series.amiUsage.reporter(influx);
   }
 
+  /** 
+   * Get the instances and spot requests for a given worker type.
+   * The return value of this is an object like:
+   * {instances: [], requests: [], deadInstances: [], deadRequests: []}
+   *
+   * If the negate parameter is true, then return information matching none
+   * of the workertypes
+   */
+  async updateWorkerTypeInRegion(ec2, workerTypes, negate = false) {
+    assert(workerTypes);
+    if (typeof workerTypes === 'string') {
+      workerTypes = [workerTypes];
+    }
+    assert(Array.isArray(workerTypes));
+
+    let instances = [];
+    let requests = [];
+    let deadInstances = [];
+    let deadRequests = [];
+
+    let keyPrefixes = workerTypes.map(workerType => negate ? '!' : '' + this.keyPrefix + workerType);
```

Comment:
This needs to be changed so that when negating, we search for a `!` of all the known workerTypes, but that we do a positive filter for `this.keyPrefix + '*'`.